interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/config/QueryTimeoutConfig.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 Alibaba Group Holding Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.graphscope.common.config;
+
+public class QueryTimeoutConfig {
+    private final long executionTimeoutMS;
+    private final long channelTimeoutMS;
+    private final long engineTimeoutMS;
+    private static final double GRADUAL_FACTOR = 0.1d;
+
+    public QueryTimeoutConfig(long executionTimeoutMS) {
+        this.executionTimeoutMS = executionTimeoutMS;
+        this.channelTimeoutMS = (long) (executionTimeoutMS * (1 - GRADUAL_FACTOR));
+        this.engineTimeoutMS = (long) (executionTimeoutMS * (1 - 2 * GRADUAL_FACTOR));
```

Comment:
Make the three time align. Should be fine.

Reply:
done